pkg/southbound/synchronizer/factory.go

```diff
@@ -42,13 +44,15 @@ func Factory(changeStore *store.ChangeStore, configStore *store.ConfigurationSto
 			ctx := context.Background()
 			completeID := utils.ToConfigName(deviceName, device.SoftwareVersion)
 			cfg := configStore.Store[store.ConfigName(completeID)]
-			mReadOnlyPaths, ok := modelReadOnlyPaths[utils.ToModelName(cfg.Type, device.SoftwareVersion)]
+			modelName := utils.ToModelName(cfg.Type, device.SoftwareVersion)
+			mReadOnlyPaths, ok := readOnlyPaths[modelName]
 			if !ok {
 				log.Warningf("Cannot check for read only paths for target %s with %s because "+
 					"Model Plugin not available - continuing", deviceName, device.SoftwareVersion)
 			}
+			operationalStateCache[deviceName] = make(map[string]*change.TypedValue)
 			sync, err := New(ctx, changeStore, configStore, &device, configChan, opStateChan,
-				errChan, mReadOnlyPaths)
+				errChan, operationalStateCache, mReadOnlyPaths)
```

Comment:
do you need to pass the cache for **all** devices down to this synchronizer. Sorry if we were through this already